library/Backends/llvm/src/Juvix/Backends/LLVM/Compilation.hs

```diff
@@ -309,6 +313,46 @@ mkPrim prim ty = case prim of
     name <- Block.generateUniqueName "LitString"
     Block.globalString (toS s) name
 
+compileRecordDecl ::
+  Types.Define m =>
+  Monad m =>
+  Types.RecordDecl ->
+  -- | Primitive value.
+  Types.Annotated Types.TermLLVM ->
+  -- | Type of the primitive.
+  m LLVM.Operand
+compileRecordDecl (recordName, fieldDecls) term = do
+  Record.register recordName (map fst fieldDecls) (map (typeToLLVM . snd) fieldDecls)
+  compileTerm term
+
+compileRecord ::
+  Types.Define m =>
+  Monad m =>
+  Types.TypeLLVM ->
+  -- | Primitive value.
+  Types.TermRecordConstructor ->
+  -- | Type of the primitive.
+  m LLVM.Operand
+compileRecord (Types.RecordType recordTypeName) (recordName, fieldTerms) | recordTypeName == recordName = do
```

Comment:
Might be best to newline the `|` here

Reply:
Done in https://github.com/anoma/juvix/pull/1106/commits/6fcae69dd44e08cbf3b1b650b307b927b9f3a93e